src/main/java/org/broadinstitute/hellbender/utils/read/ReadCoordinateComparator.java

```diff
@@ -63,22 +61,22 @@ public int compare( GATKRead first, GATKRead second ) {
         return result;
     }
 
-    private int compareCoordinates( final GATKRead first, final GATKRead second ) {
-        if ( first.isUnmapped() ) {
-            return second.isUnmapped() ? 0 : 1;
+    public static int compareCoordinates( final GATKRead first, final GATKRead second, final SAMFileHeader header ) {
+        final int firstRefIndex = ReadUtils.getAssignedReferenceIndex(first, header);
+        final int secondRefIndex = ReadUtils.getAssignedReferenceIndex(second, header);
+
+        if ( firstRefIndex == -1 ) {
+            return (secondRefIndex == -1 ? 0 : 1);
         }
-        if ( second.isUnmapped() ) {
+        else if ( secondRefIndex == -1 ) {
             return -1;
         }
 
-        final int firstRefIndex = ReadUtils.getReferenceIndex(first, header);
-        final int secondRefIndex = ReadUtils.getReferenceIndex(second, header);
-
-        final int refIndexComparison = firstRefIndex - secondRefIndex;
-        if ( refIndexComparison != 0 ) {
-            return refIndexComparison;
+        final int refIndexDifference = firstRefIndex - secondRefIndex;
+        if ( refIndexDifference != 0 ) {
+            return refIndexDifference;
```

Comment:
Code is a bit simpler this way, and also closer to `SAMRecordCoordinateComparator`, whose `fileOrderCompare()` method we are trying to imitate.